### PR TITLE
Tokens API exploration

### PR DESCRIPTION
**Pull request by aaronccasanova**
Review/Suggestions for the [new Tokens API](https://github.com/Shopify/polaris/pull/304).

### CODE REVIEW

File: packages/functions/index.ts [hunk 1]
```diff
@@ -62,105 +73,131 @@ const hslToHex = (
...
-    CSSName: getVariableName(key, 'css'),
+    figmaName: createFormatTokenName('figma', tokenName),
+    SassName: createFormatTokenName('sass', tokenName),
+    CSSName: createFormatTokenName('css', tokenName),
   };
 };
 
+type Formatter = (tokens: Tokens) => string;
+
+const CSSTokenFormatter: Formatter = (tokens) => {
```
[the remark this answers is not in the document]
Comment (aaronccasanova) [reply packages/functions/index.ts]: Great suggestion! https://github.com/Shopify/polaris/pull/321/commits/7df6b0e624046697680f615b814641b6fb8dba42

File: packages/functions/index.ts [hunk 2]
```diff
@@ -62,105 +73,131 @@ const hslToHex = (
...
+      lines.push(`${tab}${varName}: var(${alias});`);
+    }
+  });
+
+  lines.push('}');
+
+  return lines.join('\n');
+};
+
+const SASSTokenFormatter: Formatter = (tokens) => {
```
[the remark this answers is not in the document]
Comment (aaronccasanova) [reply packages/functions/index.ts]: https://github.com/Shopify/polaris/pull/321#discussion_r710470300

File: packages/functions/index.ts [hunk 3]
```diff
@@ -62,105 +73,131 @@ const hslToHex = (
...
-            const alias = getVariableName(token.aliasOf, 'sass');
-            lines.push(`${varName}: ${alias};`);
-          }
-        }
-      });
-      break;
-    }
-  }
-  return lines.join('\n');
+export const formatTokens = ({format, tokens}: FormatTokensOptions): string => {
```
[the remark this answers is not in the document]
Comment (aaronccasanova) [reply packages/functions/index.ts]: https://github.com/Shopify/polaris/pull/321#discussion_r710470300

File: packages/functions/index.ts [hunk 4]
```diff
@@ -212,57 +253,65 @@ export const getColorTokens = (): TokenList => {
...
   for (let i = 1; i <= NUMBER_OF_ITEMS_IN_100_SCALES; i++) {
-    const key = `margin-${i * 100}`;
-    const aliasKey = getSpaceTokenForIndex(i);
-    tokens[key] = {
-      aliasOf: aliasKey,
-      description: `A margin equal to ${aliasKey}`,
-      meta: createTokenMeta(key),
+    const tokenName = `margin-${i * 100}`;
+
+    const aliasTokenName = getSpaceTokenName(i);
```
[the remark this answers is not in the document]
Comment (aaronccasanova) [reply packages/functions/index.ts:286]: The variable is used in two places for both blocks and thus I would prefer not invoking the getter twice. Does this suggestion still apply? Should we update the name to make things clearer?
image

File: api/pages/api/beta/tokens.ts [hunk 5]
```diff
@@ -1,14 +1,22 @@
 import type {NextApiRequest, NextApiResponse} from 'next';
 import {formatTokens, getAllTokens} from '../../../../packages/functions';
-import {TokenList} from '../../../../packages/functions/types';
+import {Tokens} from '../../../../packages/functions/types';
 
-export const ALLOWED_FORMATS = ['json', 'css', 'sass'];
-export const DEFAULT_FORMAT = 'json';
+// SUGGESTION: This (and types) would be helpful to expose from the @polaris/tokens
+export const FORMATS = ['json', 'css', 'sass'] as const;
```
[the remark this answers is not in the document]
Comment (aaronccasanova) [reply api/pages/api/beta/tokens.ts]: Great call out 👍  I went with `SUPPORTED_FORMATS` and leveraged the `@polaris/functions` types to help ensure the API and `functions` library stay in sync: https://github.com/Shopify/polaris/commit/0a4164bb3c5bbf3900acc9568aeeaed1e1b7cac5

**Comment by aaronccasanova** [comment]
@martenbjork Per our discussion offline, I [updated every instance of `Format` to `Platform`](https://github.com/Shopify/polaris/commit/3dd1ea4c5dd262a8f2a797d71b32bac7e6d8371e) to address possible naming confusions around the term `format` being both a verb and noun. Additionally, the update to `Platform` terminology will hypothetically scale better as we [add more token creators](https://github.com/Shopify/polaris/blob/3dd1ea4c5dd262a8f2a797d71b32bac7e6d8371e/packages/functions/index.ts#L181-L203) for other platforms (i.e. `Figma`, `IOS`, `Android`, etc.)

#### Added context for any onlookers 👀 :
The term `Format` is both a verb and noun and thus utilities such as `createFormatTokenName` we're a little confusing at first glance.

Our first attempt at renaming was to put the term `Format` at the end of each utility (i.e. `convertTokensToFormat`, `convertTokenNameToFormat`) which is slightly more verbose but makes the behavior of the functions clear.

By updating our terminology to `Platform`, we avoid any confusion around multiple parts of speech (verb, noun) and keep the concise naming convention for our utilities (i.e. `createPlatformTokens`, `createPlatformTokenName`).

**Review by martenbjork — APPROVED** [review]

**Pull request merged**